Implicit conversion warning in trilinos_block_sparse_matrix

```
[620/686] Building CXX object source/lac/CMakeFiles/obj_lac_debug.dir/trilinos_block_sparse_matrix.cc.o
../source/lac/trilinos_block_sparse_matrix.cc:184:12: warning: implicit conversion turns floating-point number into integer: 'const double' to 'bool' [-Wfloat-conversion]
           drop_tolerance);
           ^~~~~~~~~~~~~~
1 warning generated.
```
This line was changed in 756c54b30c9745be729557d147634c75645e6682.
It looks like the `reinit` function with this signature is being now called:
```
  BlockSparseMatrix::reinit(
    const std::vector<IndexSet> &   parallel_partitioning,
    const BlockSparsityPatternType &block_sparsity_pattern,
    const MPI_Comm &                communicator,
    const bool                      exchange_data);
```

--- a/include/deal.II/lac/trilinos_block_sparse_matrix.h
+++ b/include/deal.II/lac/trilinos_block_sparse_matrix.h
@@ -172,6 +172,19 @@ namespace TrilinosWrappers
      * This function initializes the Trilinos matrix using the deal.II sparse
      * matrix and the entries stored therein. It uses a threshold to copy only
      * elements whose modulus is larger than the threshold (so zeros in the
+     * deal.II matrix can be filtered away).
+     */
+    void
+    reinit(
+      const std::vector<IndexSet> &              parallel_partitioning,
+      const ::dealii::BlockSparseMatrix<double> &dealii_block_sparse_matrix,
+      const MPI_Comm &                           communicator = MPI_COMM_WORLD,
+      const double                               drop_tolerance = 1e-13);
+
+    /**
+     * This function initializes the Trilinos matrix using the deal.II sparse
+     * matrix and the entries stored therein. It uses a threshold to copy only
+     * elements whose modulus is larger than the threshold (so zeros in the
      * deal.II matrix can be filtered away). Since no Epetra_Map is given, all
      * the elements will be locally stored.
      */

--- a/source/lac/trilinos_block_sparse_matrix.cc
+++ b/source/lac/trilinos_block_sparse_matrix.cc
@@ -162,6 +162,43 @@ namespace TrilinosWrappers
 
   void
   BlockSparseMatrix::reinit(
+    const std::vector<IndexSet> &              parallel_partitioning,
+    const ::dealii::BlockSparseMatrix<double> &dealii_block_sparse_matrix,
+    const MPI_Comm &                           communicator,
+    const double                               drop_tolerance)
+  {
+    const size_type n_block_rows = parallel_partitioning.size();
+
+    Assert(n_block_rows == dealii_block_sparse_matrix.n_block_rows(),
+           ExcDimensionMismatch(n_block_rows,
+                                dealii_block_sparse_matrix.n_block_rows()));
+    Assert(n_block_rows == dealii_block_sparse_matrix.n_block_cols(),
+           ExcDimensionMismatch(n_block_rows,
+                                dealii_block_sparse_matrix.n_block_cols()));
+
+    // Call the other basic reinit function ...
+    reinit(n_block_rows, n_block_rows);
+
+    // ... and then assign the correct
+    // data to the blocks.
+    for (size_type r = 0; r < this->n_block_rows(); ++r)
+      for (size_type c = 0; c < this->n_block_cols(); ++c)
+        {
+          this->sub_objects[r][c]->reinit(parallel_partitioning[r],
+                                          parallel_partitioning[c],
+                                          dealii_block_sparse_matrix.block(r,
+                                                                           c),
+                                          communicator,
+                                          drop_tolerance);
+        }
+
+    collect_sizes();
+  }
+
+
+
+  void
+  BlockSparseMatrix::reinit(
     const ::dealii::BlockSparseMatrix<double> &dealii_block_sparse_matrix,
     const double                               drop_tolerance)
   {

